src/redux/selectors/index.js

```diff
@@ -203,6 +203,38 @@ const filteredItemsInDateRangeSelector = createSelector(
   },
 );
 
+export const selectedRecordsGroupedByTypeSelector = createSelector(
+  [filteredItemsInDateRangeSelector, activeCollectionResourceTypeSelector],
+  (items, selectedResourceType) => {
+    const typeMap = items
+      .reduce((acc, item) => {
+        const { type, subType } = item;
+        return produce(acc, (draft) => {
+          // eslint-disable-next-line no-param-reassign
+          draft[type] = draft[type] ?? {};
+          // eslint-disable-next-line no-param-reassign
+          draft[type][subType] = draft[type][subType] ?? [];
+          draft[type][subType].push(item.id);
+        });
+      }, {});
+
+    const sortedGroupedResources = Object
+      .entries(typeMap)
+      .reduce((acc, [type, subTypes]) => acc.concat({
+        type,
+        label: SINGULAR_RESOURCE_TYPES[type],
+        subTypes: Object
+          .entries(subTypes)
+          .map(([subType, recordIds]) => ({ subType, recordIds }))
+          .sort(({ subType: s1 }, { subType: s2 }) => (
+            (s1.toLowerCase() < s2.toLowerCase()) ? -1 : 1)),
+      }), [])
+      .sort(({ label: l1 }, { label: l2 }) => ((l1.toLowerCase() < l2.toLowerCase()) ? -1 : 1));
+
+    return sortedGroupedResources.filter((group) => group.type === selectedResourceType);
```

Comment:
`.filter(...)` could just follow `.sort`

Reply:
i did this because in my next pr, i'm going to break out the grouping function to use in a couple places. so this .`filter` will be unique to this selector